deploy/twingate-operator/templates/pre-delete-cleanup.yaml

```diff
@@ -0,0 +1,29 @@
+{{- if index .Values "kubernetes-access-gateway" "enabled" }}
```

Comment:
We can't write `.Values.kubernetes-access-gateway.enabled` here because `kubernetes-access-gateway` is not a standard naming convention, it's a known [design flaw](https://github.com/helm/helm/issues/9636#issuecomment-1481749407) when using sub-chart. This is a workaround I found on this [thread](https://github.com/helm/helm/issues/2192#issuecomment-306033149).

Reply:
Yeah. Looking at the [standard library](https://pkg.go.dev/text/template#hdr-Functions) used in Helm, a map will work here too.

> index
	Returns the result of indexing its first argument by the
	following arguments. Thus "index x 1 2 3" is, in Go syntax,
	x[1][2][3]. Each indexed item must be a map, slice, or array.

Reply:
Btw, there’s a weird behavior that `dig` doesn’t work on .Values (see [issue](https://github.com/helm/helm/issues/9266)) e.g. dig "kubernetes-access-gateway" "enabled" "false" .Values gives this error:
```
executing "twingate-operator/templates/NOTES.txt" at <dig "kubernetes-access-gateway" "enabled" "false" .Values>: error calling dig: interface conversion: interface {} is chartutil.Values, not map[string]interface {}
```

The workaround is use `dig "kubernetes-access-gateway" "enabled" "false"  (.Values | dict )`

Reply:
The reason we are getting `dig: interface conversion: interface {} is chartutil.Values, not map[string]interface {}` is because of the type assertion [here](https://github.com/Masterminds/sprig/blob/8cb06fe3c8b0f1163c26b0a558669da72ee14656/dict.go#L154)

Go expected a type of `map[string]interface{}` but got `chartutil.Values` instead. Though `chartutil.Values` is just a wrapper of [`map[string]interface{}`](https://github.com/gabrtv/helm/blob/eee6e143d89c55d1f481d54ba2deb891ae60446f/pkg/chartutil/values.go#L37-L38), Go expects us to convert the type explicitly.

Reply:
I also checked `index` returns nil (evaluate to `false`) when `kubernetes-access-gateway.enabled` key is missing. On another note, we must use `index` in `line 2` because `dig` traverses another level down (`Subcharts.kubernetes-access-gateway.<additional_key>`) but we only needed first level (`Subcharts.kubernetes-access-gateway`).